File: src/test/java/dev/daae/time/repositories/SessionRepositoryTest.java
package dev.daae.time.repositories;

import static org.assertj.core.api.Assertions.assertThat;

import dev.daae.time.IntegrationTest;
import dev.daae.time.models.Session;
import java.time.LocalDateTime;
import java.time.ZoneOffset;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;

class SessionRepositoryTest extends IntegrationTest {

    @Autowired
    private SessionRepository repository;

    @Test
    void testThatLogIsReturnedFromRepositoryAfterItIsSaved() {
        var now = LocalDateTime.now().atOffset(ZoneOffset.UTC);
        var session = repository.save(Session.builder().start(now).end(now).build());
        repository.findById(session.getId()).orElseThrow();
    }

    @Test
    void testThatTimestampIsEqualBeforeAndAfterStoringIt() {
        var timestamp = LocalDateTime.now().atOffset(ZoneOffset.UTC);
        var id = repository.save(Session.builder().start(timestamp).build()).getId();
        var session = repository.findById(id).orElseThrow();
        assertThat(session.getStart()).isEqualTo(timestamp);
    }
}

File: src/test/kotlin/dev/daae/time/repositories/SessionRepositoryTest.kt
package dev.daae.time.repositories

import dev.daae.time.models.Session
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Disabled
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import java.time.LocalDateTime
import java.time.ZoneOffset

internal class SessionRepositoryTest(
    @Autowired
    private var repository: SessionRepository,
) : dev.daae.time.IntegrationTest() {
    @Test
    fun testThatLogIsReturnedFromRepositoryAfterItIsSaved() {
        val now = LocalDateTime.now().atOffset(ZoneOffset.UTC)
        val session = repository.save<Session>(Session(null, now, now))
        session.id?.let { repository.findById(it).orElseThrow() }
    }

    @Test
    @Disabled // TODO: Fix timezone issue
    fun testThatTimestampIsEqualBeforeAndAfterStoringIt() {
        val timestamp = LocalDateTime.now().atOffset(ZoneOffset.UTC)
        val id = repository.save(Session(null, timestamp, null)).id
        val session: Session = id?.let { repository.findById(it).orElseThrow() }!!
        Assertions.assertThat(session.start).isEqualTo(timestamp)
    }
}
